kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -322,10 +322,11 @@ private fun String.escapeDollarSigns(): String {
 private fun TypeName.unwrapTypeAlias(): TypeName {
   return when (this) {
     is ClassName -> {
-      tag<TypeNameAliasTag>()?.type ?: this
+      tag<TypeNameAliasTag>()?.type?.copy(nullable = isNullable, annotations = annotations) ?: this
```

Comment:
I'm not sure this does the right thing for e.g. 

```kotlin
typealias Foo = Bar?
data class Bar(val whatever: Int)

data class Test(val foo: Foo)
```

`isNullable` for `Foo` is `false`. When resolving `Foo` to `Bar?` the value is `true`. The `copy()` then overwrites the correctly resolved value with `false`. (I may be wrong. I haven't tested this.)

Reply:
Oh that's a good catch. Tried a few cases

```kotlin
data class Bar(val whatever: Int)
typealias Foo = Bar?

// foo is a nullable Bar
data class Test(val foo: Foo?) // <- will lint `?` as redundant
```

```kotlin
data class Bar(val whatever: Int)
typealias Foo = Bar

// foo is a nullable Bar
data class Test(val foo: Foo?)
```

```kotlin
data class Bar(val whatever: Int)
typealias Foo = Bar?

// foo is a nullable Bar
data class Test(val foo: Foo)
```

So all three of the above cases, my takeaway is that we should mark this as nullable if either `foo's` declared or underlying types are nullable. Does that sound right?

Reply:
Implemented in 6d131d1 and test in b3fffe2